econml/orf/_ortho_forest.py

```diff
@@ -1300,6 +1322,81 @@ def effect_inference(self, X=None, *, T0=0, T1=1):
                                       output_names=self._estimator.cate_output_names(),
                                       treatment_names=self._estimator.cate_treatment_names())
 
+    def _marginal_effect_inference_helper(self, T, X):
+        if self._estimator.treatment_featurizer is None:
+            return self.const_marginal_effect_inference(X)
+
+        X, T = check_input_arrays(X, T)
+        X, T = self._estimator._expand_treatments(X, T, transform=False)
+
+        feat_T = self._estimator.transformer.transform(T)
+
+        jac_T = self._estimator.transformer.jac(T)
+
+        params, cov = zip(*(self._predict_wrapper(X)))
+        params = np.array(params)
+        cov = np.array(cov)
+
+        eff_einsum_str = 'mf, mtf-> mt'
+
+        # conditionally expand jacobian dimensions to align with einsum str
+        jac_index = [slice(None), slice(None), slice(None)]
+        if ndim(T) == 1:
+            jac_index[1] = None
+        if ndim(feat_T) == 1:
+            jac_index[2] = None
+
+        # Calculate the effects
+        eff = np.einsum(eff_einsum_str, params, jac_T[tuple(jac_index)])
+
+        # Calculate the standard deviations for the effects
+        d_t_orig = T.shape[1:]
+        d_t_orig = d_t_orig[0] if d_t_orig else 1
+        self.d_t_orig = d_t_orig
+        output_shape = [X.shape[0]]
+        if T.shape[1:]:
+            output_shape.append(T.shape[1])
+        scales = np.zeros(shape=output_shape)
+
+        for i in range(d_t_orig):
+            # conditionally index multiple dimensions depending on shapes of T, Y and feat_T
+            jac_index = [slice(None)]
+            me_index = [slice(None)]
+            if T.shape[1:]:
+                jac_index.append(i)
+                me_index.append(i)
+
+            if feat_T.shape[1:]:  # if featurized T is not a vector
+                jac_index.append(slice(None))
+            else:
+                jac_index.append(None)
+
+            mid = np.einsum('mj, mjk -> mk', jac_T[tuple(jac_index)], cov)
+            final = np.einsum('mk, mk -> m', mid, jac_T[tuple(jac_index)])
```

Comment:
I think you could do this is a single call to einsum and it would be easier to follow.  (But please double-check my logic!)

```suggestion
            jac = jac_T[tuple(jac_index)]
            final = np.einsum('mj, mjk, mk -> m', jac, cov, jac)
```